include/swift/Frontend/Frontend.h

```diff
@@ -76,6 +76,11 @@ class CompilerInvocation {
 
   CodeCompletionCallbacksFactory *CodeCompletionFactory = nullptr;
 
+  /// \brief Used for holding the Swift code passed into the command line via
+  /// the "-e" flag.
+  // SR-5860 WIP: commented out for now because it produces a compiler error
+  //  std::unique_ptr<llvm::MemoryBuffer> ImmediateExecutionBuffer;
```

Comment:
@jrose-apple you'll have to forgive me but the last time I used C++ on a regular basis was quite a long time ago, well before C++11, so I'm afraid I'm in a bit over my head here.

When this line is uncommented, the `llvm:MemoryBuffer` above produces this compiler error: 
> Object of type 'swift::CompilerInvocation' cannot be assigned because its copy assignment operator is implicitly deleted

The error is produced against line 75 in Frontend.cpp, where it's assigning `CompilerInstance`'s `Invocation` property to that which is passed in to the `setup` method.

After searching around a bit I'm not sure of the best way to deal with this.

Reply:
See [Deleted implicitly declared copy constructors](http://en.cppreference.com/w/cpp/language/copy_constructor).  When you added a `unique_ptr` as a member, it deleted the copy constructor for CompilerInvocation because it now owns a resource that is `move`-only.  

Not really sure what to do about this, but maybe this indicates adding a singular `MemoryBuffer` to CompilerInvocation isn't the right abstraction.  Were you hoping to copy all the executable strings together into one contiguous buffer owned by the invocation?

Reply:
Yeah, that was the plan.

I suppose we could have `CompilerInstance` own the buffer containing those "-e" code snippets. It already contains a couple of `unique_ptr`s so it should be safe to add another, if I'm not mistaken.

Let me play around with this a bit and see how to make this work. Thanks for the help!

Reply:
Hm. I have no problem with CompilerInstance owning this, but unfortunately I think we need the buffer constructed during argument parsing, which means the CompilerInstance shouldn't come into play. (One CompilerInvocation ought to be able to set up multiple CompilerInstances.) But CompilerInvocation is already fairly expensive to copy, so using a `std::shared_ptr` doesn't seem like the end of the world for now.

A longer-term answer might be to *not* have CompilerInvocation be copyable, and to have `CompilerInstance::setup` take an rvalue reference instead of a constant reference, so that it can acquire ownership. But then we'd have to hunt down all callers of that method, and I don't want to put that on you for this first patch.

Reply:
OK, let me give that a go, using a `std::shared_ptr` on `CompilerInvocation`. Thanks!